From 2c20412176fba85effad915fb065bc13dbc1cff2 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:58 -0400
Subject: [tools] perf/hists: Do not pass period and weight to add_hist_entry()

Message-id: <1408953210-26343-171-git-send-email-jolsa@redhat.com>
Patchwork-id: 88891
O-Subject: [PATCH RHEL7.1 BZ1133083 170/322] perf hists: Do not pass period and weight to add_hist_entry()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f1cbf78d175e6202a29f53a7f915520e40a37baf
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Wed Dec 18 14:21:11 2013 +0900

description
===========
The @entry argument already has the info so no need to pass them.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index 822903e..63234e3 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -342,15 +342,15 @@ static u8 symbol__parent_filter(const struct symbol *parent)
 }
 
 static struct hist_entry *add_hist_entry(struct hists *hists,
-          struct hist_entry *entry,
-          struct addr_location *al,
-          u64 period,
-          u64 weight)
+      struct hist_entry *entry,
+      struct addr_location *al)
 {
  struct rb_node **p;
  struct rb_node *parent = NULL;
  struct hist_entry *he;
  int64_t cmp;
+ u64 period = entry->stat.period;
+ u64 weight = entry->stat.weight;
 
  p = &hists->entries_in->rb_node;
 
@@ -437,7 +437,7 @@ struct hist_entry *__hists__add_entry(struct hists *hists,
   .transaction = transaction,
  };
 
- return add_hist_entry(hists, &entry, al, period, weight);
+ return add_hist_entry(hists, &entry, al);
 }
 
 int64_t
-- 
1.7.1